ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/MessageScrubber.java

```diff
@@ -7,36 +7,18 @@
 import com.google.common.collect.Maps;
 import com.google.common.collect.Streams;
 import com.mozilla.telemetry.ingestion.core.Constant.Attribute;
-import com.mozilla.telemetry.transforms.MapElementsWithErrors.MessageShouldBeDroppedException;
 import java.util.Map;
 import java.util.Optional;
 import java.util.stream.Stream;
 import org.apache.beam.sdk.metrics.Metrics;
 
+/**
+ * This class is called in {@link ParsePayload} to check for known signatures of potentially
+ * harmful data. The {@link #scrub(Map, ObjectNode)} method may throw an exception as a signal
+ * to route the message to error output or to be dropped.
+ */
 public class MessageScrubber {
 
-  private enum ScrubAction {
-    DROP, REDACT, SEND_TO_ERRORS
-  }
```

Comment:
We remove this enum and instead build metrics emission into the exceptions, so that the individual cases have explicit `throw` calls. The overall effect feels more legible to me with this change, but I'm definitely open to different opinions.